Notify/Notify.Functions/Notify.Functions/NotifyFunctions/Users/UpdateProfilePicture.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Newtonsoft.Json;
+using Notify.Functions.Core;
+using Notify.Functions.HTTPClients;
+
+namespace Notify.Functions.NotifyFunctions.Users
+{
+    public static class UpdateProfilePicture
+    {
+        [FunctionName("UpdateProfilePicture")]
+        [AllowAnonymous]
+        public static async Task<IActionResult> RunAsync(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "put", "post", Route = "user/profilePicture")]
+            HttpRequest req, ILogger log)
+        {
+            IMongoCollection<BsonDocument> collection;
+            string requestBody;
+            dynamic data;
+            FilterDefinition<BsonDocument> filter;
+            UpdateDefinition<BsonDocument> update;
+            ObjectResult result;
+
+            log.LogInformation("Received request to update user profile picture");
+
+            try
+            {
+                collection = AzureDatabaseClient.Instance.GetCollection<BsonDocument>(Constants.DATABASE_NOTIFY_MTA,
+                    Constants.COLLECTION_USER);
+
+                requestBody = await new StreamReader(req.Body).ReadToEndAsync();
+                data = JsonConvert.DeserializeObject(requestBody);
+
+                filter = Builders<BsonDocument>.Filter.Eq("userName", Convert.ToString(data.userName));
+                update = BuildUpdateDefinition(data);
+
+                if (update == null)
+                {
+                    log.LogWarning("No updates provided.");
+                    return new BadRequestObjectResult("No updates provided.");
```

Comment:
Specific to this case I need to return because the next line updates the DB with what I need to update. If you still want me to change I will refactor the flow of the method.